pkg/protocol/snapshotcreator/snapshotcreator.go

```diff
@@ -71,7 +70,15 @@ func CreateSnapshot(opts ...options.Option[Options]) error {
 	for _, accountData := range opt.Accounts {
 		// Only add genesis validators if an account has both - StakedAmount and StakingEndEpoch - specified.
 		if accountData.StakedAmount > 0 && accountData.StakingEpochEnd > 0 {
-			accounts.Set(blake2b.Sum256(accountData.IssuerKey[:]), &account.Pool{
+
+			blockIssuerKeyEd25519, ok := accountData.IssuerKey.(iotago.BlockIssuerKeyEd25519)
```

Comment:
It doesn't need to be of type ed25519 I guess.
We could also directly hash the IssuerKeyBytes.